proc_snmp6.go

```diff
@@ -0,0 +1,381 @@
+	OutMcastPkts     float64
+	InOctets         float64
+	OutOctets        float64
+	InMcastOctets    float64
+	OutMcastOctets   float64
+	InBcastOctets    float64
+	OutBcastOctets   float64
+	InNoECTPkts      float64
+	InECT1Pkts       float64
+	InECT0Pkts       float64
+	InCEPkts         float64
+}
+
+type Icmp6 struct {
+	InMsgs                    float64
+	InErrors                  float64
+	OutMsgs                   float64
+	OutErrors                 float64
+	InCsumErrors              float64
+	InDestUnreachs            float64
+	InPktTooBigs              float64
+	InTimeExcds               float64
+	InParmProblems            float64
+	InEchos                   float64
+	InEchoReplies             float64
+	InGroupMembQueries        float64
+	InGroupMembResponses      float64
+	InGroupMembReductions     float64
+	InRouterSolicits          float64
+	InRouterAdvertisements    float64
+	InNeighborSolicits        float64
+	InNeighborAdvertisements  float64
+	InRedirects               float64
+	InMLDv2Reports            float64
+	OutDestUnreachs           float64
+	OutPktTooBigs             float64
+	OutTimeExcds              float64
+	OutParmProblems           float64
+	OutEchos                  float64
+	OutEchoReplies            float64
+	OutGroupMembQueries       float64
+	OutGroupMembResponses     float64
+	OutGroupMembReductions    float64
+	OutRouterSolicits         float64
+	OutRouterAdvertisements   float64
+	OutNeighborSolicits       float64
+	OutNeighborAdvertisements float64
+	OutRedirects              float64
+	OutMLDv2Reports           float64
+	InType1                   float64
+	InType134                 float64
+	InType135                 float64
+	InType136                 float64
+	InType143                 float64
+	OutType133                float64
+	OutType135                float64
+	OutType136                float64
+	OutType143                float64
+}
+
+type Udp6 struct {
+	InDatagrams  float64
+	NoPorts      float64
+	InErrors     float64
+	OutDatagrams float64
+	RcvbufErrors float64
+	SndbufErrors float64
+	InCsumErrors float64
+	IgnoredMulti float64
+}
+
+type UdpLite6 struct {
+	InDatagrams  float64
+	NoPorts      float64
+	InErrors     float64
+	OutDatagrams float64
+	RcvbufErrors float64
+	SndbufErrors float64
+	InCsumErrors float64
+}
+
+func (p Proc) Snmp6() (ProcSnmp6, error) {
+	filename := p.path("net/snmp6")
+	data, err := util.ReadFileNoStat(filename)
+	if err != nil {
+		// On systems with IPv6 disabled, this file won't exist.
+		// Do nothing.
+		if errors.Is(err, os.ErrNotExist) {
+			return ProcSnmp6{PID: p.PID}, nil
+		}
+
+		return ProcSnmp6{PID: p.PID}, err
+	}
+
+	procSnmp6, err := parseSNMP6Stats(bytes.NewReader(data))
+	procSnmp6.PID = p.PID
+	return procSnmp6, err
+}
+
+// parseSnmp6 parses the metrics from proc/<pid>/net/snmp6 file
+// and returns a map contains those metrics.
+func parseSNMP6Stats(r io.Reader) (ProcSnmp6, error) {
+	var (
+		scanner  = bufio.NewScanner(r)
```

Comment:
How did this gofmt issue not surfaced in the PR? @SuperQ @pgier any idea?

Reply:
I will fix this fmt issue and also try to investigate why this did worked properly

Reply:
Weird. Maybe out of date source branch?